src/lib.rs

```diff
@@ -120,12 +120,39 @@ impl<Kind, Type> IdMap<Kind, Type> where Type: Eq + Clone, Kind: Clone {
     }
 }
 
-type SyncExtSender = Mutex<Box<ExtSender<WatchEvent>>>;
-type WatchersMap = HashMap<usize, Arc<SyncExtSender>>;
+#[derive(Debug, Copy, Clone, PartialEq)]
+enum SendDirection {
```

Comment:
Nit: Not sure that `SendDirection` is the right name, but I have no better to offer.